config/routes.rb

```diff
@@ -3,4 +3,14 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get 'users/', to: 'users#index'
+
+  get 'users/:id', to:'users#show'
+
+  get 'users/:user_id/posts', to:'posts#index'
+
+  get 'users/:user_id/posts/:id', to:'posts#show'
+ 
```

Comment:
- [ **OPTIONAL** ] I think it would be better if you use Rails `resources` notation as it doesn't require a lot of code and it is very easy as you only need to specify the action you want in your routes. For more details about `resources` notation, I suggest you refer this **[article](https://launchschool.com/books/demystifying_rails/read/routes_and_resources#resourcenotation)**. All the above can be updated to the below ones:
```ruby
  root 'users#index'

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
@tresorsawasawa  What constructive feedback it is! Thank you, bro

but the root default is already set 
```ruby

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```